### ISSUE TITLE
compile error

### ISSUE THREAD

**Issue by skulko** [avujic/xmr_arch64#6]
I cannot build it anymore because compile fails (it was working at the beginning of the month). Now I get these errors when building on rpi3:

`/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:303:76: warning: multi-character character constant [-Wmultichar]
             << "  " << rapidjson::GetParseError_En(doc.GetParseError()) << '\n"';
                                                                            ^~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:29:1: error: ‘include’ does not name a type
 include <getopt.h>
 ^~~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:124:1: error: elements of array ‘const option options []’ have incomplete type
 };
 ^
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:124:1: error: storage size of ‘options’ isn’t known
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:146:1: error: elements of array ‘const option config_options []’ have incomplete type
 };
 ^
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:146:1: error: storage size of ‘config_options’ isn’t known
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:157:1: error: elements of array ‘const option pool_options []’ have incomplete type
 };
 ^
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:157:1: error: storage size of ‘pool_options’ isn’t known
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:165:1: error: elements of array ‘const option api_options []’ have incomplete type
 };
 ^
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:165:1: error: storage size of ‘api_options’ isn’t known
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In constructor ‘Options::Options(int, char**)’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:224:67: error: ‘getopt_long’ was not declared in this scope
         key = getopt_long(argc, argv, short_options, options, NULL);
                                                                   ^
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:229:28: error: ‘optarg’ was not declared in this scope
         if (!parseArg(key, optarg)) {
                            ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:234:9: error: ‘optind’ was not declared in this scope
     if (optind < argc) {
         ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:254:21: error: ‘Cpu’ has not been declared
         m_threads = Cpu::optimalThreadsCount(m_algo, m_doubleHash, m_maxCpuUsage);
                     ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:257:27: error: ‘Cpu’ has not been declared
         const int count = Cpu::optimalThreadsCount(m_algo, m_doubleHash, m_maxCpuUsage);
                           ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In member function ‘bool Options::getJSON(const char*, rapidjson::Document&)’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:302:9: error: ‘cout’ is not a member of ‘std’
         std::cout << fileName << "  " << (int) doc.GetErrorOffset() \
         ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In member function ‘bool Options::parseArg(int, const char*)’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:400:25: error: ‘Cpu’ has not been declared
             m_threads = Cpu::threads();
                         ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In member function ‘void Options::parseJSON(const option*, const Value&)’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:620:16: error: invalid use of incomplete type ‘const struct option’
     if (!option->name || !object.HasMember(option->name)) {
                ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:620:50: error: invalid use of incomplete type ‘const struct option’
     if (!option->name || !object.HasMember(option->name)) {
                                                  ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:624:50: error: invalid use of incomplete type ‘const struct option’
     const rapidjson::Value &value = object[option->name];
                                                  ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:626:15: error: invalid use of incomplete type ‘const struct option’
     if (option->has_arg && value.IsString()) {
               ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:627:24: error: invalid use of incomplete type ‘const struct option’
         parseArg(option->val, value.GetString());
                        ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:629:20: error: invalid use of incomplete type ‘const struct option’
     else if (option->has_arg && value.IsUint64()) {
                    ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:630:24: error: invalid use of incomplete type ‘const struct option’
         parseArg(option->val, value.GetUint64());
                        ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:632:21: error: invalid use of incomplete type ‘const struct option’
     else if (!option->has_arg && value.IsBool()) {
                     ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:633:28: error: invalid use of incomplete type ‘const struct option’
         parseBoolean(option->val, value.IsTrue());
                            ^~
In file included from /opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:35:0:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.h:37:8: note: forward declaration of ‘struct option’
 struct option;
        ^~~~~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In member function ‘int Options::getAlgoVariant() const’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:715:16: error: ‘Cpu’ has not been declared
         return Cpu::hasAES() ? AV1_AESNI : AV3_SOFT_AES;
                ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:718:20: error: ‘Cpu’ has not been declared
     if (m_safe && !Cpu::hasAES() && m_algoVariant <= AV2_AESNI_DOUBLE) {
                    ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp: In member function ‘int Options::getAlgoVariantLite() const’:
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:730:16: error: ‘Cpu’ has not been declared
         return Cpu::hasAES() ? AV2_AESNI_DOUBLE : AV4_SOFT_AES_DOUBLE;
                ^~~
/opt/cpuminer-multi-avujic/xmr_arch64/src/Options.cpp:733:20: error: ‘Cpu’ has not been declared
     if (m_safe && !Cpu::hasAES() && m_algoVariant <= AV2_AESNI_DOUBLE) {
                    ^~~
make[2]: *** [CMakeFiles/xmr_arch64.dir/build.make:471: CMakeFiles/xmr_arch64.dir/src/Options.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/xmr_arch64.dir/all] Error 2
make: *** [Makefile:84: all] Error 2`

**Comment by avujic** [comment]
In Options.cpp line 29
just add # in from of include <getopt.h>

It is my bad, will be changed in repository ASAP.

**Comment by avujic** [comment]
Please check and confirm.

**Comment by ordtrogen** [comment]
Here are my results (git pull + rebuild)

TLDR: updated from Git, build still fails: 

/home/rutger/xmr_arch64/src/net/Url.h:69:10: error: 'string' in namespace 'std' does not name a type


![image](https://user-images.githubusercontent.com/15184875/38024915-c0da2fc4-3286-11e8-86c4-2684baf2ee47.png)


![image](https://user-images.githubusercontent.com/15184875/38024930-cf241fe0-3286-11e8-9cab-e22c3eb6a9ff.png)

**Comment by avujic** [comment]
make
Scanning dependencies of target xmr_arch64

[ 88%] Building C object CMakeFiles/xmr_arch64.dir/src/crypto/c_jh.c.o
[ 90%] Building C object CMakeFiles/xmr_arch64.dir/src/crypto/c_skein.c.o
[ 92%] Building CXX object CMakeFiles/xmr_arch64.dir/src/crypto/CryptoNight.cpp.o
[ 95%] Building CXX object CMakeFiles/xmr_arch64.dir/src/log/SysLog.cpp.o
[ 97%] Building CXX object CMakeFiles/xmr_arch64.dir/src/api/Httpd.cpp.o
[100%] Linking CXX executable xmr_arch64
[100%] Built target xmr_arch64

**Comment by ordtrogen** [comment]
Now it works for me. Thanks

**Comment by avujic** [comment]
Good. You are very welcome.
